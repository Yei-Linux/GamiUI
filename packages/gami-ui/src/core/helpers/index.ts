export * from './date-formatter'
export * from './validations.helper'
