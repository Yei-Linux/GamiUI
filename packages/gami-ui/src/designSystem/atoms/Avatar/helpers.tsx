import { TextModeType } from 'core/domain/types'
import { avatarTextModes } from './constants'
import { REGEX_RULES } from 'core/utils/constants'

export const validateMaxLetters = (
  firstLettersGot: string,
  maxSizeLetters: number
) => {
  const size = firstLettersGot.length

  if (size > maxSizeLetters) {
    return firstLettersGot.substring(0, maxSizeLetters)
  }

  return firstLettersGot
}

export const showOnlyFirstLettersOnText = (
  text: string,
  type: TextModeType,
  maxSizeLetters: number
): string => {
  const regexType = avatarTextModes[type]
  const firstLettersCaptured: string[] | null = text.match(
    REGEX_RULES[regexType]
  )

  const joinFirstLetters = firstLettersCaptured?.join('') || ''

  return validateMaxLetters(joinFirstLetters, maxSizeLetters)
}
