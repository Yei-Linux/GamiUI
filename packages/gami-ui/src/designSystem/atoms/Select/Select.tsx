import { CSSObject } from '@emotion/serialize'
import React from 'react'
import { OptionProps, OptionTypeBase } from 'react-select'
import { lightTheme } from 'styles/tokens/lightTheme'

import { IInput } from '../Input/Input'
import * as S from './Select.styles'

export interface IOptions {
  value: string
  label: string | React.ReactNode
}

type TSelectValue = OptionTypeBase[] | OptionTypeBase | null | undefined

export interface ISelect extends Omit<IInput, 'value'> {
  /**
   * Children Element
   */
  options?: IOptions[]
  /**
   * Multiple Option
   */
  isMultiple?: boolean
  /**
   * isClearable Option
   */
  isClearable?: boolean
  /**
   * isClearable Option
   */
  value?: TSelectValue
}

const Select = ({
  options,
  onChangeFormItem,
  placeholder,
  value,
  isMultiple = false,
  isClearable = false,
}: ISelect) => {
  const colourStyles = {
    option: (
      styles: CSSObject,
      { isSelected, isFocused }: OptionProps<any, any, any>
    ): CSSObject => {
      const getBgOption = () => {
        if (isSelected) return lightTheme.secondary.jordyBlueMedium
        if (isFocused) return lightTheme.secondary.jordyBlueLight
        return 'transparent'
      }

      return {
        ...styles,
        backgroundColor: getBgOption(),
      }
    },
  }

  return (
    <S.ReactSelect
      isClearable={isClearable}
      isMulti={isMultiple}
      classNamePrefix="Select"
      placeholder={placeholder}
      value={value}
      onChange={onChangeFormItem}
      options={options}
      styles={colourStyles}
    />
  )
}

export default Select
