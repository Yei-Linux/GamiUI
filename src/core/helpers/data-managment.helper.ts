export const mapEnum = (enumerable: any): any[] => {
  const enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  )
  return enumMembers
}
