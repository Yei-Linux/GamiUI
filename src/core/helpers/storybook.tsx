import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};