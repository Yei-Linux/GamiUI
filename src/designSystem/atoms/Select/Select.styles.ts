import styled from "@emotion/styled";
import {
  SlideTopAnimation,
  SlideBottomAnimation,
} from "../../../styles/utilities/transitions";

export const SelectWrapper = styled.div`
  position: relative;
  width: fit-content;
`;

export const Selectinput = styled.div`
  width: fit-content;
`;

export const SelectOptionsWrapper = styled.div<{ isToggle: boolean }>`
  z-index: 99;
  overflow-y: auto;
  max-height: 200px;
  top: 50px;
  position: absolute;
  background: #fafafa;
  box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.12);
  border-radius: 5px;
  padding: 0px 20px;
  width: 100%;

  ${(props) => (props.isToggle ? SlideTopAnimation : SlideBottomAnimation)}
`;

export const OptionWrapper = styled.div`
  padding: 10px 0px;
  &:hover {
    cursor: pointer;
  }
`;
