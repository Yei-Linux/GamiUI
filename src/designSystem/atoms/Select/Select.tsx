import React, { useContext, useEffect, useRef, useState } from "react";
import SelectProvider from "../../../context/SelectProvider";
import context from "../../../context/SelectProvider/context";
import useClickOutside from "../../../hooks/useClickOutside";
import Input from "../Input";
import { IInput } from "../Input/Input";
import {
  OptionWrapper,
  Selectinput,
  SelectOptionsWrapper,
  SelectWrapper,
} from "./Select.styles";

export interface SelectProps extends IInput {
  /**
   * Children Element
   */
  children?: React.ReactNode[];
  /**
   * Is basic select
   */
  isBasic?: boolean;
}

const SelectContent = ({ children, isBasic, ...args }: SelectProps) => {
  const selectReference: any = useRef();
  const [isToggle, setIsToggle] = useState(false);

  const { optionSelected } = useContext(context);

  useClickOutside({
    ref: selectReference,
    handleDoAction: () => setIsToggle(false),
  });

  useEffect(()=>{
    args.onChangeFormItem(optionSelected)
  },[optionSelected])

  const handleInputClick = (): void => setIsToggle(!isToggle);

  return (
    <SelectWrapper ref={selectReference}>
      <Selectinput onClick={handleInputClick}>
        <Input readOnly={isBasic} {...args} value={optionSelected?.text} />
      </Selectinput>
      <SelectOptionsWrapper isToggle={isToggle}>
        {children}
      </SelectOptionsWrapper>
    </SelectWrapper>
  );
};

const Select = ({ children, isBasic, ...args }: SelectProps) => {
  return (
    <SelectProvider>
      <SelectContent children={children} isBasic={isBasic} {...args} />
    </SelectProvider>
  );
};

Select.Option = ({ children, value }: { children: string; value: string }) => {
  const { setOptionSelected } = useContext(context);
  const handleClickOption = () => setOptionSelected({ value, text: children });
  return <OptionWrapper onClick={handleClickOption}>{children}</OptionWrapper>;
};

Select.defaultProps = {
  isBasic: true,
  width: "NORMAL",
  heigth: "SMALL",
  border: "MEDIUM",
};

export default Select;
