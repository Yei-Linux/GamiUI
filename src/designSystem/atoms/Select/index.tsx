export { default } from './Select'
export { options } from './constants'