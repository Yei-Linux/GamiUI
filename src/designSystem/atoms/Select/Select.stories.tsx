import React from "react";
import { Meta } from "@storybook/react";
import { stylesControl } from "../../../core/helpers/storybook";

import Select from ".";

export default {
  title: "Atoms/Select",
  component: Select,
  args: { ...Select.defaultProps },
  argTypes: {
    placeholder: { control: "text" },
    ...stylesControl,
  },
} as Meta;

export const BasicSelect = (args: any) => {
  return (
    <Select {...args}>
      <Select.Option value="one" >Option1</Select.Option>
      <Select.Option value="two">Option2</Select.Option>
      <Select.Option value="three">Option3</Select.Option>
    </Select>
  );
};
BasicSelect.args = {
  placeholder: "Type your option",
  width: "NORMAL",
  heigth: "SMALL",
  shadow: "",
  border: "MEDIUM",
};
