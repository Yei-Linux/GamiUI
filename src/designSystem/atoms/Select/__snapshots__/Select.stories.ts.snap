// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Select Basic Select 1`] = `
<div
  className="css-htg7a1-SelectWrapper ee0eyh30"
>
  <div
    className="css-1dhedq7-Selectinput ee0eyh31"
    onClick={[Function]}
  >
    <input
      autoComplete="off"
      className="css-gwomp1-InputContainer e7vlt9n1"
      name="gamification"
      onChange={[Function]}
      placeholder="Type your option"
      prefix={null}
      readOnly={true}
      type="input"
      value=""
      width="NORMAL"
    />
  </div>
  <div
    className="css-ae753p-SelectOptionsWrapper ee0eyh32"
  >
    <div
      className="css-u0k5sv-OptionWrapper ee0eyh33"
      onClick={[Function]}
    >
      Option1
    </div>
    <div
      className="css-u0k5sv-OptionWrapper ee0eyh33"
      onClick={[Function]}
    >
      Option2
    </div>
    <div
      className="css-u0k5sv-OptionWrapper ee0eyh33"
      onClick={[Function]}
    >
      Option3
    </div>
  </div>
</div>
`;
